### PR TITLE
Initialize i18n module earlier

### PR DESCRIPTION
**Pull request by tk0miya**
This allows to translate some warning messages on parsing conf.py.

### CODE REVIEW

File: sphinx/locale/__init__.py [hunk 1]
```diff
@@ -229,11 +229,17 @@ def lazy_gettext(string):
 if PY3:
     def _(message):
         # type: (unicode) -> unicode
-        return translators['sphinx'].gettext(message)
+        try:
+            return translators['sphinx'].gettext(message)
+        except KeyError:
+            return message
```
Comment (shimizukawa) [inline sphinx/locale/__init__.py:235]: Why this change is needed?
Did the change of the initialization timing have any effect?

**Review by shimizukawa — APPROVED** [review]
LGTM with nits

File: sphinx/locale/__init__.py (hunk 1 above)
Comment (tk0miya) [reply sphinx/locale/__init__.py:235]: This is needed for `Config.convert_overrides()`. It is called both before and after `_init_i18n()`. The KeyError might be raised only if user gives invalid values to pre-init config values.

Comment (shimizukawa) [reply sphinx/locale/__init__.py:235]: I understand. Thanks.

**Comment by tk0miya** [comment]
Thank you for reviewing! Merged
